registry/handlers/app.go

```diff
@@ -249,6 +257,14 @@ func (app *App) dispatcher(dispatch dispatchFunc) http.Handler {
 			context.Repository = notifications.Listen(
 				repository,
 				app.eventBridge(context, r))
+
+			for _, mw := range app.Config.Middleware["repository"] {
+				rmw, err := repositorymiddleware.Get(mw.Name, mw.Options, context.Repository)
+				if err != nil {
+					panic(fmt.Sprintf("unable to configure repository middleware (%s): %s", mw.Name, err))
```

Comment:
I agree that panicking doesn't make sense. It's too bad there's no easy way to fail fast up front when starting up, however. Since the registry administrator presumably expects the middleware to load correctly, maybe something like a 500 server error would be appropriate?

Sent from my iPhone

> On Mar 8, 2015, at 12:56 AM, David Lawrence <notifications@github.com> wrote:
> 
> In registry/handlers/app.go:
> 
> > @@ -249,6 +257,14 @@ func (app *App) dispatcher(dispatch dispatchFunc) http.Handler {
> >  			context.Repository = notifications.Listen(
> >  				repository,
> >  				app.eventBridge(context, r))
> > +
> > +			for _, mw := range app.Config.Middleware["repository"] {
> > +				rmw, err := repositorymiddleware.Get(mw.Name, mw.Options, context.Repository)
> > +				if err != nil {
> > +					panic(fmt.Sprintf("unable to configure repository middleware (%s): %s", mw.Name, err))
> Agreed. @ncdc How would you expect this to react?
> 
> —
> Reply to this email directly or view it on GitHub.
>